contracts/integrations/custom/CustomIntegrationIdleFinance.sol

```diff
@@ -0,0 +1,338 @@
+        uint256[] calldata /* _minAmountsOut */
+    )
+        internal
+        pure
+        override
+        returns (
+            address,
+            uint256,
+            bytes memory
+        )
+    {
+        address vault = BytesLib.decodeOpDataAddress(_data);
+        bytes memory methodData = abi.encodeWithSelector(IIdleToken.redeemIdleToken.selector, _resultTokensIn);
+
+        return (vault, 0, methodData);
+    }
+
+    /**
+     * The list of addresses of the IERC-20 tokens mined as rewards during the strategy
+     *
+     * hparam  _data                      Address provided as param
+     * @return address[] memory           List of reward token addresses
+     */
+    function _getRewardTokens(
+        address /* _data */
+    ) internal pure override returns (address[] memory) {
+        // No extra rewards. All get consolidated to the vault input token on exit
+        return new address[](1);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * The tokens to be purchased by the strategy on enter according to the weights.
+     * Weights must add up to 1e18 (100%)
+     *
+     * hparam  _data                      Address provided as param
+     * @return _inputTokens               List of input tokens to buy
+     * @return _inputWeights              List of weights for the tokens to buy
+     */
+    function getInputTokensAndWeights(
+        bytes calldata _data
+    ) external view override returns (address[] memory _inputTokens, uint256[] memory _inputWeights) {
+        // Vault takes 100% in the vault token
+        address vault = BytesLib.decodeOpDataAddress(_data);
+        address[] memory inputTokens = new address[](1);
+        inputTokens[0] = IIdleToken(vault).token();
+        uint256[] memory inputWeights = new uint256[](1);
+        inputWeights[0] = 1e18; // 100%
+        return (inputTokens, inputWeights);
+    }
+
+    /**
+     * The tokens to be received on exit.
+     *
+     * hparam  _strategy                  Strategy address
+     * hparam  _data                      Bytes data
+     * hparam  _liquidity                 Number with the amount of result tokens to exit
+     * @return exitTokens                 List of output tokens to receive on exit
+     * @return _minAmountsOut             List of min amounts for the output tokens to receive
+     */
+    function getOutputTokensAndMinAmountOut(
+        address _strategy,
+        bytes calldata _data,
+        uint256 /* _liquidity */
+    ) external view override returns (address[] memory exitTokens, uint256[] memory _minAmountsOut) {
+        // Vault exits 100% in the vault token
+        address vault = BytesLib.decodeOpDataAddress(_data);
+        address[] memory outputTokens = new address[](1);
+        outputTokens[0] = IIdleToken(vault).token();
+        uint256[] memory outputAmounts = new uint256[](1);
+        outputAmounts[0] = IIdleToken(vault).tokenPriceWithFee(_strategy);
```

Comment:
They have both `tokenPrice()` and `tokenPriceWithFee(address user)` methods, I am pretty sure we want the latter. Is `_strategy` the right address to pass?

Reply:
You still need to multiply this price by the amount of liquidity that is deposited right?

Reply:
Makes sense, thanks!